content/2016.01.loadtesting.with-oauth-part-2.rst

```diff
@@ -0,0 +1,264 @@
+`Requests Authentication <http://docs.python-requests.org/en/latest/user/authentication/>`_
+to ease the use of Firefox Account Bearer Token with
+`requests <http://docs.python-requests.org/en/latest/>`_ and 
+`httpie <http://httpie.org>`_.
+
+With this plugin, we can basically provide our user credentials and
+the plugin will build a valid Bearer Token for our request:
+
+.. code-block:: bash
+
+    http https://profile.accounts.firefox.com/v1/profile \
+        --auth-type fxa-bearer \
+        --auth "email@domain.tld:password"
+
+
+Using fxa-client to generate Bearer Tokens
+------------------------------------------
+
+The code we wrote at the time was quite similar to what we were trying
+to do for the load-test setup: Generating a Bearer Token for a
+given user.
+
+We ended-up building a CLI tool called ``fxa-client`` that is able to
+generate a BASH file that export some environment variables:
+
+.. code-block:: bash
+
+    USAGE: fxa-client [-h] [--bearer] [--create] [--auth AUTH]
+                      [--out OUTPUT_FILE] [--verbose] [--user-salt FXA_USER_SALT]
+                      [--prefix PREFIX] [--account-server ACCOUNT_SERVER_URL]
+                      [--oauth-server OAUTH_SERVER_URL] [--client-id CLIENT_ID]
+                      [--scopes SCOPES]
+
+With this client, instead of building a Bearer Token for the request,
+we can just ask the ``fxa-client`` to build one for our load-test:
+
+.. code-block:: bash
+
+    fxa-client --auth email@domain.tld --bearer --scopes "profile"
+
+    Please enter a password for email@domain.tld: 
+    # ---- BEARER TOKEN INFO ----
+    # User: email@domain.tld
+    # Scopes: profile
+    # Account: https://api-accounts.stage.mozaws.net/v1
+    # Oauth: https://oauth.stage.mozaws.net/v1
+    # Client ID: 5882386c6d801776
+    # ---------------------------
+    export OAUTH_BEARER_TOKEN="b1d4babc01502fa8d9fc0139757168bf5908f33abc66f46253d4c69468e39373"
+
+Then we can use this Bearer Token for our request:
+
+.. code-block:: bash
+
+    http https://profile.stage.mozaws.net/v1/profile \
+        Authorization:"Bearer b1d4babc01502fa8d9fc0139757168bf5908f33abc66f46253d4c69468e39373"
+
+
+Using fxa-client to create new test accounts
+--------------------------------------------
+
+We can also create a random user (on the stage environment only) and
+generate a Bearer Token for it:
+
+.. code-block:: bash
+
+    fxa-client --create --bearer --prefix my-app
+
+    # ---- BEARER TOKEN INFO ----
+    # User: my-app-6318a65dde1efc2f4c3f7b4e6cb33188@restmail.net
+    # Scopes: profile
+    # Account: https://api-accounts.stage.mozaws.net/v1
+    # Oauth: https://oauth.stage.mozaws.net/v1
+    # Client ID: 5882386c6d801776
+    # ---------------------------
+    export OAUTH_BEARER_TOKEN="90abc87ed1621ee504c1252ed382abc8269d1abc29f2ff87cc5e25f00249fabc"
+
+If we want to reuse the same account multiple time and create it only
+if needed, we can provide the user-salt (as a valid base64 string):
```

Comment:
You need to rephrase this sentence: "To avoid creating extraneous user accounts, it is possible to specify a user-salt [...]"

You need to explain what the user salt would be used for.